kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h

```diff
@@ -977,12 +1020,38 @@ class ResidualBasedNewtonRaphsonStrategy
                 {
                     if (GetKeepSystemConstantDuringIterations() == false)
                     {
-                        //A = 0.00;
                         TSparseSpace::SetToZero(rA);
                         TSparseSpace::SetToZero(rDx);
                         TSparseSpace::SetToZero(rb);
 
                         p_builder_and_solver->BuildAndSolve(p_scheme, r_model_part, rA, rDx, rb);
+                        if (damped) {
```

Comment:
Still, I think we should refactor the CC to avoid code duplication and flexibility